validator/lib/validateCmpt.js

```diff
@@ -0,0 +1,136 @@
+    }
+
+    var byteOffset = 16;
+    var sizeOfUint32 = 4;
+    var isValid = true;
+
+    var magic = content.toString('utf8', 0, 4);
+    var version = content.readUInt32LE(4);
+    var byteLength = content.readUInt32LE(8);
+    var tilesLength = content.readUInt32LE(12);
+
+    if (magic !== 'cmpt') {
+        return {
+            result : false,
+            message: 'Cmpt header has an invalid magic field. Expected magic = \'cmpt\'. Found magic = ' + magic
+        };
+    }
+
+    if (version !== 1) {
+        return {
+            result : false,
+            message: 'Cmpt header has an invalid version field. Expected version = 1. Found version = ' + version
+        };
+    }
+
+    if (byteLength !== content.length) {
+        return {
+            result : false,
+            message: 'Cmpt header has an invalid byteLength field. Expected byteLength = ' + content.length + '. Found byteLength = ' + byteLength
+        };
+    }
+
+    for (var i = 0; i < tilesLength; i++) {
+        var errorAddon = '';
+        var innerTileMagic = content.toString('utf8', byteOffset, byteOffset + sizeOfUint32);
+        var innerTileByteLength = content.readUInt32LE(byteOffset + 2 * sizeOfUint32);
+
+        var innerTile;
+        var validatorResult;
+
+        if (innerTileMagic === 'b3dm') {
+            innerTile = new Buffer(24);
+            content.copy(innerTile, 0, byteOffset, byteOffset + innerTileByteLength);
+            validatorResult = validateB3dm(innerTile);
+            isValid = isValid && validatorResult.result;
+        }
+
+        else if (innerTileMagic === 'i3dm'){
+            innerTile = new Buffer(32);
+            content.copy(innerTile, 0, byteOffset, byteOffset + innerTileByteLength);
+            validatorResult = validateI3dm(innerTile);
+            isValid = isValid && validatorResult.result;
+        }
+
+        else if (innerTileMagic === 'pnts'){
+            innerTile = new Buffer(28);
+            content.copy(innerTile, 0, byteOffset, byteOffset + innerTileByteLength);
+            validatorResult = validatePnts(innerTile);
+            isValid = isValid && validatorResult.result;
+        }
+
+        else if (innerTileMagic === 'cmpt') {
+            innerTile = new Buffer(innerTileByteLength);
+            content.copy(innerTile, 0, byteOffset, byteOffset + innerTileByteLength);
+            validatorResult = validateCmpt(innerTile);
+            isValid = isValid && validatorResult.result;
+        }
+
+        else {
+            isValid = false;
+            errorAddon += 'Inner tile header magic cannot be identified';
+        }
+
+        byteOffset = byteOffset + innerTileByteLength; // skip over this tile
+
+        if( (byteOffset + 12 > byteLength) && (i + 1 !== tilesLength) ) {
+            isValid = false;
+            errorAddon += 'Cmpt header given tilesLength = ' + tilesLength + '. Found number of inner tiles = ' + (i + 1);
+        }
```

Comment:
The check 
```
if (byteOffset !== byteLength) {
        return {
            result : false,
            message: 'Header has invalid inner tile formats or tileLength field. Expected end of tile = ' + byteLength + '. Found end of tile = ' + byteOffset
        };
    }
```
checks the condition where all inner tiles are valid, but there are extra bytes left over. I didn't see a more readable way to combine it with the other check indicated above. I did move the other check, but the only way I can see to properly check this is to check it after the for-loop finishes because that's when we know all inner tiles are valid.